File: src/network/packet/impl/request/PlayerNotificationCheckRequestPacket.php
<?php

namespace pocketcloud\cloud\network\packet\impl\request;

use pocketcloud\cloud\cache\impl\NotificationListCache;
use pocketcloud\cloud\network\client\ServerClient;
use pocketcloud\cloud\network\packet\impl\response\PlayerNotificationCheckResponsePacket;
use pocketcloud\cloud\network\packet\RequestPacket;
use pocketcloud\cloud\network\packet\util\PacketData;

final class PlayerNotificationCheckRequestPacket extends RequestPacket {

    public function __construct(private string $player = "") {}

    public function handle(ServerClient $client): void {
        $this->sendResponse(PlayerNotificationCheckResponsePacket::create(NotificationListCache::is($this->player)), $client);
    }

    public function decodePayload(PacketData $packetData): void {
        $packetData->readAll($this->player);
    }

    public function getPlayer(): string {
        return $this->player;
    }

    public static function create(string $player): self {
        return new self($player);
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/network/packet/impl/request/PlayerNotificationCheckRequestPacket.java
package de.pocketcloud.cloud.network.packet.impl.request;

import de.pocketcloud.common.cache.LocalCache;
import de.pocketcloud.cloud.cache.NotificationListCache;
import de.pocketcloud.cloud.network.client.ServerClient;
import de.pocketcloud.cloud.network.packet.RequestPacket;
import de.pocketcloud.cloud.network.packet.impl.response.PlayerNotificationCheckResponsePacket;
import de.pocketcloud.network.packet.data.PacketData;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.jetbrains.annotations.NotNull;

@NoArgsConstructor
@Getter
public final class PlayerNotificationCheckRequestPacket extends RequestPacket {

    private String player;

    public PlayerNotificationCheckRequestPacket(String player) {
        this.player = player != null ? player : "";
    }

    @Override
    public void handle(@NotNull ServerClient client) {
        sendResponse(PlayerNotificationCheckResponsePacket.create(LocalCache.get(NotificationListCache.class).contains(player)), client);
    }

    @Override
    public void decodePayload(PacketData packetData) {
        this.player = packetData.readString();
    }

    public static PlayerNotificationCheckRequestPacket create(String player) {
        return new PlayerNotificationCheckRequestPacket(player);
    }
}